src/vm/gdbjit.cpp

```diff
@@ -2138,8 +2315,25 @@ bool NotifyGdb::BuildSectionTable(MemBuf& buf)
             pSh->sh_entsize = 0;
     }
 
+    for (int i = SectionNamesCount; i < SectionNamesCount + thunks_count; ++i, ++pSh)
+    {
+        char name[256];
+        sprintf(name, ".thunk_%i", i);
```

Comment:
It looks like even more unification would be beneficial from the same reason I've mentioned in my previous comment. 
It looks like the safest way would be to merge these two functions together so that you would update the pSh at the same time you generate the names into the buffer.